python_modules/libraries/dagster-postgres/dagster_postgres/run_storage/run_storage.py

```diff
@@ -248,9 +248,13 @@ def _order_by_run_priority(self, query: SqlAlchemyQuery) -> SqlAlchemyQuery:
                 db.func.coalesce(
                     # try to get the priority tag value
                     db.func.cast(
-                        db.func.cast(RunsTable.c.run_body, db.JSON)
-                        .op("->")("tags")
-                        .op("->>")(PRIORITY_TAG),
+                        db.func.regexp_replace(
+                            db.func.cast(RunsTable.c.run_body, db.JSON)
+                            .op("->")("tags")
+                            .op("->>")(PRIORITY_TAG),
+                            r"^(?!-?\d+$).*",
+                            "0",
+                        ),
```

Comment:
big yikes - spinning up regex definitely seems like it might be risky on the CPU cost front if theres a large set of candidates

Reply:
https://github.com/dagster-io/dagster/pull/18175 as a potential alternative